### ISSUE TITLE
Reintroduce plotting loss per training step?

### ISSUE THREAD

**Issue by vpratz** [bayesflow-org/bayesflow#512]
In v1, we stored the metrics for each training step, and offered an option for plotting this higher resolution in the diagnostics `loss` function. This functionality is currently broken, as Keras by default only stores the metrics at the end of each batch, but still advertised in the function signature. A user [recently reached out](https://discuss.bayesflow.org/t/plotting-of-the-loss/168) asking for this functionality, so I think we should either provide it, or remove the argument from the diagnostics function.

Storing the metrics can be done with a custom callback (below for completion). I personally do not think it to be super important, and showing how to do it with a custom callback in the docs might be enough, but would be glad to hear other opinions as well. What do you think?

<details><summary>Callback code</summary>

```python
class StepHistory(keras.callbacks.Callback):
    """Callback that records events into a `StepHistory` object in every batch."""

    def __init__(self):
        super().__init__()
        self.history = {}

    def on_train_batch_end(self, batch, logs=None):
        keys = list(logs.keys())
        for k, v in logs.items():
            self.history.setdefault(k, []).append(v)

    def on_epoch_end(self, epoch, logs=None):
        # Set the history attribute on the model after the epoch ends. This will
        # make sure that the state which is set is the latest one.
        self.model.step_history = self
```

</details>

**Comment by stefanradev93** [comment]
I also think that this is not important. There is really very little value of storing the loss(es) at each training step and requires more boilerplate. Adding an example / hint to the docs should be enough.

**Comment by vpratz** [comment]
I have removed the parameter in 212af40e2, as it was never functional in v2 I think we do not need a proper deprecation.
